src/builder/builder.go

```diff
@@ -254,33 +254,53 @@ func (b *Builder) AddBundles(bundles []string, force bool, git bool) int {
 		b.UpdateRepo(b.Clearver, false)
 	}
 
-	var includes []string
-	for _, bundle := range bundles {
-		// Check if bundle exists in clrbundledir
-		if _, err := os.Stat(clrbundledir + bundle); os.IsNotExist(err) {
-			helpers.PrintError(errors.New("Bundle " + bundle + " does not exist in CLR version " + b.Clearver))
+	// Copy all bundles over into mix-bundles if -all passed
+	if allbundles {
+		files, err := ioutil.ReadDir(clrbundledir)
+		if err != nil {
+			helpers.PrintError(err)
 			os.Exit(1)
 		}
-		// Check if bundle exists in mix bundles dir
-		if _, err := os.Stat(bundledir + bundle); os.IsNotExist(err) || force {
-			// Parse bundle to get all includes
-			if ib, err := helpers.GetIncludedBundles(clrbundledir + bundle); err != nil {
-				helpers.PrintError(errors.New("Cannot parse bundle " + bundle + " from CLR version " + b.Clearver))
+		bundles = make([]string, len(files))
+		for _, file := range files {
+			// Check if bundle exists in mix bundles dir
+			if _, err := os.Stat(bundledir + file.Name()); os.IsNotExist(err) || force {
+				helpers.CopyFile(bundledir+file.Name(), clrbundledir+file.Name())
```

Comment:
there is no handling of the error code returned by CopyFile. Same thing on the block below.

Reply:
This is an excellent catch. I followed the example in the UpdateRepos function, where it also doesn't check the error. I think I will fix all of these locations in a separate commit on this pull request.

Reply:
Yeah, I also noticed it in other places. I'm adding the error handling for all users of CopyFile as part of my init patch that I'll post later today. I'm also changing it a bit to transfer the error handler to the caller.

Reply:
+1 for moving the error printing out of the helper libraries and into the calling functions -- that's the whole point of returning an error to the caller. I know @cmarcelo will cheer too.

If you can, please separate general code cleanup out of your feature enhancement commits. Separation of concerns. I'd be happy with them being separate commits in the same PR, but @tmarcu might want a cleaner divide.